game/src/select_cards.rs

```diff
@@ -49,13 +52,343 @@ fn get_crib_cards(hand: &[Card], held_cards: &[Card]) -> Vec<Card> {
+                }
+            }
+            _ => {}
+        }
+    }
 
+    //
+    //  if we get here - we have no cards that we can play that give points, no runs we can try to start, and no pairs to leverage.
+    //  so we will try to pick a card assuming that the other player only has cards of rank 10 to play left
+    //
+    //  Another innovation idea to test is that a human player will figure out what the game is always doing -- so, for example, it is easy
+    //  to figure out that if the computer starts with (say) a 8, then they are probably trying to match a pair.  if they start with a card
+    //  less than 5, they are trying to get to 15.  this extra information can be leveraged to help the player.  So instead we should figure
+    //  out what valid cards could be played and then if then set strategic_weight to be the same for adding up to 15 and for pairs and then
+    //  randomly pick which one to use.
+    //
+
+    if strategic_weight > 0 {
+        if card_to_play.rank == Rank::Unknown {
+            panic!("strategic weight is set but the card_to_play is not not");
+        }
+        return Ok(card_to_play);
+    }
+    //
+    //  if the last card (the highest value) is not a 5, return it
+    //
+    if cards_left[cards_left.len() - 1].value != 5 {
+        return Ok(cards_left[cards_left.len() - 1]);
+    }
+    //
+    //  if it is a 5, return the next highest one -- we know it is not a pair because if it was, we'd have played it based on what we have above
+    //  we also know that we have at least two cards, because if there was only one, we would have already picked it.
+    Ok(cards_left[cards_left.len() - 2])
+}
+#[cfg(test)]
+mod tests {
+    use crate::cards::{Card, Rank::*, Suit as Of};
+    use card as c;
     use super::*;
 
-    #[test]
+    macro_rules! test_case {
+        (
+    $name:ident,
+    $counted_cards:expr,
+    $cards_left:expr,        
+    $expected_card:expr,   
+    $expected_go:literal,             
+
+    ) => {
+            #[test]
+            fn $name() {
+                let result = super::get_next_counted_card($counted_cards, $cards_left);
+                match result {
+                    Err(e) => {
+                        println!("an error returned: {:?}", e);
+                        if ($expected_go == false) {
+                            assert!($expected_go, "Unexpected Go!");
+                        }
+                    }
+                    Ok(card) => {
+                        println!("card returned: {}", card.name());
+                        assert_eq!(
+                            card.name(),
+                            $expected_card.name(),
+                            "unexpected card returned: {}.  expected: {}",
+                            card.name(),
+                            $expected_card.name()
+                        );
+                    }
+                }
+            }
+        };
+    }
+    /*
```

Comment:
blank line

we should get consistent on comment style. If we wanted to go with `///`, does Rust do the right thing with a Doc comment over a macro like this?